### ISSUE TITLE
Pruner: allow setting different "keep" and "keepSince" settings per namespace

### ISSUE THREAD

**Issue by wilstdu** [tektoncd/operator#1921]
### Feature request

Allow setting different "keep" and "keepSince" pruner settings per namespace, because namespaces may have a different amount of traffic, so in some cases it makes sense to have a certain amount of PipelineRuns in the namespace (like the namespace has a lot of traffic, many PipelineRuns created daily/hourly), and there are cases when it makes more sense to use keepSince and delete by age (namespaces that have less traffic, would keep irrelevant (old) PipelineRuns until "keep" threshold is reached - doesn't make sense, because more PipelineRuns = slower tekton-controller). Even better would be to have an ability to combine both of these settings per namespace, but that's not yet possible because of Tekton CLI (https://github.com/tektoncd/cli/issues/2206). 

### Use case

Tekton runs PipelineRuns in multiple, dynamically created namespaces, some namespaces get more traffic, others - less (for understanding: production and development branch namespace), this means that different namespaces may need different pruning options based on the workloads they run.

**Comment by tomalexander** [comment]
Is this possible using the namespace annotations? https://tekton.dev/docs/operator/tektonconfig/#pruner-namespace-annotations

**Comment by jkandasa** [comment]
@tomalexander yes, it is possible to use the namespace annotations. Please follow the document you have mentioned
